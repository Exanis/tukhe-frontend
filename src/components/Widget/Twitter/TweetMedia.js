import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';


export const classes = theme => ({
    tweetMedia: {
        minHeight: 80,
        minWidth: 80,
        flex: 1,
        backgroundSize: 'cover',
        display: "block"
    },
});

export class RawTweetMedia extends Component {
    render () {
        return <a
            onClick={this.props.onMediaClick}
            className={this.props.classes.tweetMedia}
            style={{backgroundImage: `url(${this.props.media.media_url})`}}
        />;
    }
}

const TweetMedia = withStyles(classes)(RawTweetMedia);

TweetMedia.propTypes = {
    onMediaClick: PropTypes.func.isRequired,
    media: PropTypes.shape({
        media_url: PropTypes.string.isRequired
    }).isRequired
};
RawTweetMedia.propTypes = {
    ...TweetMedia.propTypes,
    classes: PropTypes.object.isRequired
};

export default TweetMedia;
