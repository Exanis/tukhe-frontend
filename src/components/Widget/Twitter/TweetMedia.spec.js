import React from 'react';
import { RawTweetMedia, classes } from './TweetMedia';
import { shallow } from 'enzyme';

const props = {
    onMediaClick: jest.fn(),
    media: {
        media_url: 'url'
    },
    classes: {}
};

it('Should render as expected', () => {
    const wrapper = shallow(<RawTweetMedia {...props} />);

    expect(wrapper).toMatchSnapshot();
});

it('Should keep styling consistant', () => {
    expect(classes()).toMatchSnapshot();
});
