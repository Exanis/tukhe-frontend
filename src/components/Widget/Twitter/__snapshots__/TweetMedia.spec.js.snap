// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should keep styling consistant 1`] = `
Object {
  "tweetMedia": Object {
    "backgroundSize": "cover",
    "display": "block",
    "flex": 1,
    "minHeight": 80,
    "minWidth": 80,
  },
}
`;

exports[`Should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawTweetMedia
    classes={Object {}}
    media={
      Object {
        "media_url": "url",
      }
    }
    onMediaClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "onClick": [MockFunction],
      "style": Object {
        "backgroundImage": "url(url)",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "onClick": [MockFunction],
        "style": Object {
          "backgroundImage": "url(url)",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
